pkg/controllers/provisioningv2/rke2/machineprovision/args.go

```diff
@@ -134,6 +134,7 @@ func (h *handler) getArgsEnvAndStatus(infra *infraObject, args map[string]interf
 		filesSecret = constructFilesSecret(driver, args)
 		cmd = append(cmd, toArgs(driver, args, rancherCluster.Status.ClusterName)...)
 	} else {
+		filesSecret = constructFilesSecret(driver, args)
```

Comment:
I think we should be OK with moving this call above the `if` as it's exactly the same in both branches and doesn't seem to depend on anything executed in "then" portion of the if.